trade.lic

```diff
@@ -0,0 +1,259 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#trade
+=end
+custom_require.call(%w[common common-items common-money common-travel drinfomon equipmanager])
+
+class Trade
+  include DRC
+  include DRCT
+  include DRCI
+  include DRCM
+
+  def initialize
+    @town_data = get_data('town')
+    @equipment_manager = EquipmentManager.new
+    settings = get_settings
+    @contract_container = settings.trade_contract_container
+    @hometown = settings.hometown
+    bput("open my #{@contract_container}", 'You open', 'That is already')
+    @equipment_manager.empty_hands
+    @caravan = OpenStruct.new
+    recall_caravan
+  end
+
+  def run
+    unless exists?('contract')
+      nearest_minister = find_closest_id('trade_minister')
+      get_new_contract(nearest_minister)
+    end
+    contract = parse_contract
+    unless contract.presented
+      load_caravan(contract.origin_clerk)
+      recall_caravan
+      walk_to(contract.origin_outpost)
+    end
+    feed_caravan
+    take_caravan_to(contract.destination_outpost)
+    bput('get my contract', 'You get')
+    deliver_contract(contract.destination_clerk)
+    pay_dues(contract.destination_clerk)
+    deposit_coins(500, Room.current.id) if contract.destination_town == @hometown
+  end
+
+  def debugging?
+    UserVars.trade_debug
+  end
+
+  def deposit_coins(withdraw_amt, return_to_id)
+    return if wealth(@hometown) < withdraw_amt
+
+    walk_to(@town_data[@hometown]['deposit']['id'])
+    case bput('deposit all', 'you drop all your', 'You hand the clerk some coins', "You don't have any", 'There is no teller here', 'reached the maximum balance I can permit')
+    when 'There is no teller here'
+      return
```

Comment:
Do you need to `walk_to(return_to_id)` if you fail to find a teller?

Reply:
Yea, that sounds better. It was a copy and paste from sell-loot i believe.